packages/sentinel/src/api/index.ts

```diff
@@ -222,4 +225,70 @@ export class SentinelClient extends BaseApiClient {
     };
   }
+
+  private toCreateBlockSentinelRequest(sentinel: CreateBlockSubscriberResponse): CreateBlockSentinelRequest {
+    const rule = sentinel.addressRules[0];
+    let txCondition;
+
+    for (const condition of rule.conditions) {
+      for (const cond of condition.txConditions) {
+        if (cond.expression) txCondition = cond.expression;
+      }
+    }
+
+    return {
+      type: 'BLOCK',
+      addresses: rule.addresses, // There's only one addressRules at the moment, may cause problems if we add multiple address rules
+      abi: rule.abi,
+      eventConditions: _.flatten(rule.conditions.map((condition) => condition.eventConditions)),
+      functionConditions: _.flatten(rule.conditions.map((condition) => condition.functionConditions)),
+      txCondition,
+      name: sentinel.name,
+      paused: sentinel.paused,
+      alertThreshold: sentinel.alertThreshold,
+      notifyConfig: sentinel.notifyConfig,
+      autotaskCondition: rule.autotaskCondition?.autotaskId,
+      autotaskTrigger: sentinel.notifyConfig?.autotaskId,
+      alertTimeoutMs: sentinel.alertThreshold?.windowSeconds,
+      notificationChannels: sentinel.notifyConfig?.notifications?.map(({ notificationId }) => notificationId) ?? [],
+      network: sentinel.network,
+      confirmLevel: parseInt(_.last(sentinel.blockWatcherId.split('-')) as string), // We're sure there is always a last number if the convention is followd
+    };
+  }
+
+  private toCreateFortaSentinelRequest(sentinel: CreateFortaSubscriberResponse): CreateFortaSentinelRequest {
+    return {
+      type: 'FORTA',
+      name: sentinel.name,
+      paused: sentinel.paused,
+      alertThreshold: sentinel.alertThreshold,
+      notifyConfig: sentinel.notifyConfig,
+      autotaskCondition: sentinel.fortaRule.autotaskCondition?.autotaskId,
+      autotaskTrigger: sentinel.notifyConfig?.autotaskId,
+      alertTimeoutMs: sentinel.alertThreshold?.windowSeconds,
+      notificationChannels: sentinel.notifyConfig?.notifications?.map(({ notificationId }) => notificationId) ?? [],
+      network: sentinel.network,
+      fortaLastProcessedTime: sentinel.fortaLastProcessedTime,
+      addresses: sentinel.fortaRule.addresses,
+      agentIDs: sentinel.fortaRule.agentIDs,
+      fortaConditions: sentinel.fortaRule.conditions,
+    };
+  }
+
+  private toCreateSentinelRequest(sentinel: CreateSentinelResponse): CreateSentinelRequest {
+    if (sentinel.type === 'BLOCK') return this.toCreateBlockSentinelRequest(sentinel);
+    if (sentinel.type === 'FORTA') return this.toCreateFortaSentinelRequest(sentinel);
+
+    throw new Error(`Invalid sentinel type. Type must be FORTA or BLOCK`);
+  }
+
+  private mergeApiSentinelWithUpdateSentinel(
+    apiSentinel: CreateSentinelResponse,
+    sentinel: UpdateSentinelRequest,
+  ): Promise<CreateSubscriberRequest> {
+    return this.constructSentinelRequest({
+      ...this.toCreateSentinelRequest(apiSentinel),
+      ...(sentinel as CreateSentinelRequest),
```

Comment:
I'm curious why we need this explicit cast. I'd suggest we remove it an see what the compiler complains about, which could be a legitimate error we've missed!

Reply:
I gave it a second try and I could get it to work the way Dylan was proposing, but it makes me nervous that I'm doing a casting and a `any` type usage (which warns me on VSCode). [This is the snippet](https://github.com/OpenZeppelin/defender-client/pull/94/files#diff-a1360494c37bf5715a6861a8aedd3a82ee6004a19542bb90ff88da8040988385R291-R294)

However, I'll go for that alternative as you and Dylan strongly advocated for it instead of the `CreateSentinelRequest` cast. And now that there are tests, I'm confident it's working the same as with the cast

My understanding is that we have a mix of:

1. Converting client model to API model back and forth
2. We're using the `CreateSentinelRequest` type for both updating and creating
3. The `.get` endpoint returns a `CreateSentinelResponse`

It works but tbh, I think we've set up a nonaccurate type system around the Sentinel client, which is tech debt that will continue to slow down development for this client. I don't have enough knowledge to refactor it, so for now it's okay.